### PR TITLE
Split @iov/tendermint-types

### PR DESCRIPTION
**Pull request by webmaster128**
Everything from #514 except the package renaming.

The only 7 base types shared between tendermint and BCP now are

```ts
export declare enum Algorithm {
    Ed25519 = "ed25519",
    Secp256k1 = "secp256k1"
}
export declare type PublicKeyBytes = Uint8Array & As<"public-key">;
export interface PublicKeyBundle {
    readonly algo: Algorithm;
    readonly data: PublicKeyBytes;
}
/** Used to differentiate a blockchain. Should be alphanumeric or -_/ and unique */
export declare type ChainId = string & As<"chain-id">;
export declare type SignatureBytes = Uint8Array & As<"signature">;
export declare type PostableBytes = Uint8Array & As<"postable">;
export declare type TxId = Uint8Array & As<"txid">;
```

### CODE REVIEW

File: packages/iov-bcp-types/src/bcp.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Stream } from "xstream";
 
-import { ChainId, PostableBytes, PublicKeyBundle, Tag, TxId, TxQuery } from "@iov/tendermint-types";
+import { ChainId, PostableBytes, PublicKeyBundle, TxId } from "@iov/tendermint-types";
```
Comment (ethanfrey) [inline packages/iov-bcp-types/src/bcp.ts]: Weren't you going to rename `@iov/tendermint-types` to `@iov/base-types`?

File: packages/iov-bns/src/util.ts [hunk 2]
```diff
@@ -103,3 +105,15 @@ export function isSwapRelease(
     tx.transaction.kind === TransactionKind.SwapClaim || tx.transaction.kind === TransactionKind.SwapTimeout
   );
 }
+
+export function buildTxQuery(query: BcpTxQuery): QueryString {
```
Comment (ethanfrey) [inline packages/iov-bns/src/util.ts]: Yes, this is a better place for the logic

**Comment by ethanfrey** [comment]
Also, please set the project on the right hand sidebar, so it shows up in the issue tracker

**Comment by webmaster128** [comment]
> Did you change your mind, or is that a different pr?

The renaming would increase the diff dramatically so I decided to do it separately. This keeps the important stuff and the dumb stuff separated for easier review.

**Pull request merged**